code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -153,12 +153,12 @@
 	if(.)
 		for(var/obj/item/pickaxe/drill/cyborg/D in R.module)
 			R.module.remove_module(D, TRUE)
-		for(var/obj/item/shovel/S in R.module)
-			R.module.remove_module(S, TRUE)
-
-		var/obj/item/pickaxe/drill/cyborg/diamond/DD = new /obj/item/pickaxe/drill/cyborg/diamond(R.module)
-		R.module.basic_modules += DD
-		R.module.add_module(DD, FALSE, TRUE)
+			//Remove the Shovel too
+			for(var/obj/item/shovel/S in R.module)
+				R.module.remove_module(S, TRUE)
+			var/obj/item/pickaxe/drill/cyborg/diamond/DD = new /obj/item/pickaxe/drill/cyborg/diamond(R.module)
+			R.module.basic_modules += DD
+			R.module.add_module(DD, FALSE, TRUE)
```

Comment:
this is wrong now
this goes through a loop for every drill the borg has -> does a loop for every shovel **_inside_** the drill loop and adds a new drill for every other drill

the new drill for every other drill is not that bad in my opinion since they only every have one as far as i know
but the shovel loop inside the drill loop is a concern in my opinion